ImageData/FileLoader.cc

```diff
@@ -82,31 +82,60 @@ void FileLoader::FindCoords(int& spectral_axis, int& stokes_axis) {
     }
 }
 
-bool FileLoader::FindShape(IPos& shape, size_t& num_channels, size_t& num_stokes, int& spectral_axis, int& stokes_axis) {
+bool FileLoader::FindShape(IPos& shape, int& spectral_axis, int& stokes_axis, std::string& message) {
     if (!HasData(FileInfo::Data::Image))
         return false;
 
     shape = LoadData(FileInfo::Data::Image)->shape();
-    size_t num_dims = shape.size();
+    size_t _num_dims = shape.size();
```

Comment:
Does this not mask the `_num_dims` attribute with a local variable with the same name?

Reply:
I don't understand the question.  Rather than assign the local num_dims then later assign the _num_dims attribute to the local, why not just assign the attribute?

Reply:
What I mean is, are you not declaring a new local variable here, rather than using the attribute, because of the `size_t`?

Reply:
Yes, you are right!

Reply:
In a future refactoring pass I'd like to make this function no longer return values to the caller *and* set state on the loader, and also try to avoid some of the property duplication between the loader and the frame. But I don't think that needs to happen in this PR.

Reply:
Agree, DRY (don't repeat yourself)!  So then have loaders for all image types slice the data for spatial profiles and point spectral profiles, rather than doing it in the frame?

Reply:
The dividing line between loader and frame is fuzzy; I think there are various tasks that could be described as "loading" and moved into the loader, even if they're not format-specific and end up in the parent class.

Right now there's a bunch of stuff in the frame that probably doesn't need to be there -- I think it would be a good idea to discuss how to divide up responsibility between the different classes at the F2F.

Reply:
Thanks for the catch.